### PR TITLE
rpc: add channel reserve fields in RPC responses.

### LINKED ISSUE #1639: Suggestion for "channel reserve", "spendable amount", "available liquidity"
I know currently we have many issues that are priority... But if this is possible I would like to suggest for these items to be listed for each channel when we do `listchannels`: 

`channel_reserve_satoshis` = lists 1% of required channel reserve to help users aware of this fact.
`spendable_satoshis` = lists amount of money the user can spend.
`available_liquidity_satoshis` = lists available amount that can be used to route payments (which I think will be great to help users understand how routing works).

If someone can create code for these features, it would be great.  Thank you.

### PR DESCRIPTION
**Pull request by roeierez**
This PR adds two fields to channels that return as part of RPC result endpoint. For both pending and open channels the fields: local_chan_reserve, remote_chan_reserve were added.
These fields represents the reservation the nodes are required to keep in both sides of the channel.
This is provides an accurate way to calculate the "real" inbound and outbound liquidity.
If I understood correctly, I believe it is also a part of the solution to this bug in lightningapp: https://github.com/lightninglabs/lightning-app/issues/1071

Fixes #1639.

### CODE REVIEW

**Comment by Roasbeef** [comment]
Duplicate of #1998?

**Comment by roeierez** [comment]
> Duplicate of #1998?

Both indeed adds the local channel reserve field but this PR adds the remote channel reserve as well.
To me it makes sense to add both fields together.

File: lnrpc/rpc.proto [hunk 1]
```diff
@@ -1161,6 +1161,13 @@ message Channel {
 
     /// A set of flags showing the current state of the channel.
     string chan_status_flags = 19 [json_name = "chan_status_flags"];
+
+    /// This minimum satoshies this node is required to reseve in its balance    
+    int64 local_chan_reserve = 20 [json_name = "local_chan_reserve"];
```
Comment (cfromknecht) [inline lnrpc/rpc.proto]: both fields should have a `_sat` suffix to indicate the unit, same for pending channels response

nit: reseve -> reserve, also add period :)

File: lnrpc/rpc.proto [hunk 2]
```diff
@@ -1468,6 +1475,9 @@ message PendingChannelsResponse {
 
         int64 local_balance = 4 [ json_name = "local_balance" ];
         int64 remote_balance = 5 [ json_name = "remote_balance" ];
+       
+        int64 local_chan_reserve = 6 [json_name = "local_chan_reserve"];
```
Comment (cfromknecht) [inline lnrpc/rpc.proto]: can also copy the new comments to these two fields. the others _should_ be populated, but we can at least start by making incremental progress

File: rpcserver.go [hunk 3]
```diff
@@ -29,7 +29,7 @@ import (
 	"github.com/btcsuite/btcwallet/wallet/txauthor"
 	"github.com/coreos/bbolt"
 	"github.com/davecgh/go-spew/spew"
-	"github.com/grpc-ecosystem/go-grpc-middleware"
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
```
Comment (cfromknecht) [inline rpcserver.go]: doesn't look like this change is needed

File: lnrpc/rpc.proto (hunk 1 above)
Comment (roeierez) [reply lnrpc/rpc.proto]: Done.

File: lnrpc/rpc.proto (hunk 2 above)
Comment (roeierez) [reply lnrpc/rpc.proto]: Done.

File: rpcserver.go (hunk 3 above)
Comment (roeierez) [reply rpcserver.go]: Right, removed.

File: lnrpc/rpc.proto [hunk 4]
```diff
@@ -1161,6 +1161,13 @@ message Channel {
 
     /// A set of flags showing the current state of the channel.
     string chan_status_flags = 19 [json_name = "chan_status_flags"];
+
+    /// The minimum satoshies this node is required to reserve in its balance.
```
Comment (halseth) [inline lnrpc/rpc.proto]: `satoshis`?

Comment (roeierez) [reply lnrpc/rpc.proto]: Yeh.. fixed.

**Comment by cfromknecht** [comment]
@roeierez latest change shows a linter error on the travis build, o/w think this is gtg

**Comment by roeierez** [comment]
> @roeierez latest change shows a linter error on the travis build, o/w think this is gtg

fixed and also fixed my editor to run go fmt automatically.

**Comment by cfromknecht** [comment]
@roeierez needs rebase since #3279 was merged

**Comment by roeierez** [comment]
> @roeierez needs rebase since #3279 was merged

Rebased.

File: lnrpc/rpc.proto [hunk 5]
```diff
@@ -1161,6 +1161,13 @@ message Channel {
 
     /// A set of flags showing the current state of the channel.
     string chan_status_flags = 19 [json_name = "chan_status_flags"];
+
+    /// The minimum satoshis this node is required to reserve in its balance.
+    int64 local_chan_reserve_sat = 20 [json_name = "local_chan_reserve_sat"];
+
+    /// The minimum satoshis the other node is required to reserve in its 
+    /// balance.
```
Comment (wpaulino) [inline lnrpc/rpc.proto]: Multi-line comments in the protos should use `/**/` instead.

Comment (roeierez) [reply lnrpc/rpc.proto]: Fixed.

**Pull request merged**